server/endPoints-file/updateDueDateEndPoint.js

```diff
@@ -0,0 +1,48 @@
+const pool = require("../database/db");
+
+const updateDueDate = async (req, res) => {
+  try {
+    const { topicId } = req.body;
+
+    const getDueDateQuery =
```

Comment:
you are trying to check if the row exists. You have to look for a response of your UPDATE request. If the row doesn't exist the response back would be 0 rows, but if the row does exit it will return only one row with updated values
Here is a good example:
```

`app.patch("/video/:id", async (req, res) => {
  const id = req.params.id;
  const rating = parseInt(req.body.rating);

  try {
    const query = `
    UPDATE videos
    SET rating = $1
    WHERE id = $2
    RETURNING *
   `
```
```

    const request = await db.query(query, [rating, id]);
    const result = request.rows;

    // Check if any rows were affected by the update
    if (result.length === 0) {
      // If no rows were affected, it means the video with the given id wasn't found
      return res.status(404).json({ message: "Video not found" });
    }

    return res.status(200).json(result);
  } catch (err) {
    console.error(err);
    res.status(500).json({ message: "Internal server error" });
  }
});`

```